### PR TITLE
Fixing login flake

### PR DESCRIPTION
**Pull request by amitkrout**
## What is the purpose of this change? What does it change?

Making each login test spec independent.
## Was the change discussed in an issue?
https://github.com/openshift/odo/pull/2162#issuecomment-534467268

## How to test changes?

make test-cmd-login-logout

### CODE REVIEW

**Comment by kadel** [comment]
This is not a good way to fix this issue.
If there is something that depends on the ondering it should be inside one test spec.
Each test spec needs to be from each onther
/hold

**Comment by amitkrout** [comment]
> @amitkrout Shouldn't we run it close to 10 times to confirm that this patch will fix it?

I have locally tested it locally more than 10 time i guess

**Comment by amitkrout** [comment]
> This is not a good way to fix this issue.
> If there is something that depends on the ondering it should be inside one test spec.
> Each test spec needs to be from each onther
> /hold

That was my intention, but thought that it will increase the spec length and probably some early failure will cause many steps un executed. 

hmm, let put it then in a single spec.

**Comment by kadel** [comment]
> That was my intention, but thought that it will increase the spec length and probably some early failure will cause many steps un executed.
> 
> hmm, let put it then in a single spec.

ideally, we should identify what it is that makes this fail and fix it

**Comment by amitkrout** [comment]
> > That was my intention, but thought that it will increase the spec length and probably some early failure will cause many steps un executed.
> > hmm, let put it then in a single spec.
> 
> ideally, we should identify what it is that makes this fail and fix it. Not just blindly join everything together

https://github.com/openshift/odo/pull/2162#issuecomment-534467268

**Comment by amitkrout** [comment]
Failures encountered  
```
odo component command tests Creating component with numeric named context 
  should create default named component in a directory with numeric name
[...]
• Failure [112.447 seconds]
odo component command tests
/go/src/github.com/openshift/odo/tests/integration/cmd_cmp_test.go:13
  Creating component with numeric named context
  /go/src/github.com/openshift/odo/tests/integration/component.go:579
    should create default named component in a directory with numeric name []
    /go/src/github.com/openshift/odo/tests/integration/component.go:596
     Unexpected error:
      
        <*os.PathError | 0xc0000ada10>: {Op: "chdir", Path: "", Err: 0x2}
        chdir : no such file or directory
    occurred
     /go/src/github.com/openshift/odo/tests/helper/helper_filesystem.go:42
```

```
odo component command tests odo component delete, list and describe 
  should fail outside a odo directory without component name as parameter
  /go/src/github.com/openshift/odo/tests/integration/component.go:513
 [...]
[odo]  ✓  Changes successfully pushed to component


Running odo with args [odo list --project scwcuiutpy]
[odo]  ✗  There are no components deployed.

[...]
 • Failure [72.480 seconds]
odo component command tests
/go/src/github.com/openshift/odo/tests/integration/cmd_cmp_test.go:13
  odo component delete, list and describe
  /go/src/github.com/openshift/odo/tests/integration/component.go:486
    should fail outside a odo directory without component name as parameter [It]
    /go/src/github.com/openshift/odo/tests/integration/component.go:513
     Failed after 0.214s.
      
    Running odo with args [odo list --project scwcuiutpy]
    Expected
        <int>: 0
    not to match exit code:
        <int>: 0
```

```
odo component command tests odo component delete, list and describe 
  should pass outside a odo directory with component name as parameter
  /go/src/github.com/openshift/odo/tests/integration/component.go:525
 [...]
Running odo with args [odo describe nodejs --app app --project cjmwpnlhak -o json]
[odo] {"kind":"Component","apiVersion":"odo.openshift.io/v1alpha1","metadata":{"name":"nodejs","namespace":"cjmwpnlhak","creationTimestamp":null},"spec":{"app":"app","type":"nodejs","source":"file://./","env":[{"name":"DEBUG_PORT","value":"5858"}],"ports":["8080/TCP"]},"status":{"state":"Pushed"}}
[...]
 • Failure [67.511 seconds]
odo component command tests
/go/src/github.com/openshift/odo/tests/integration/cmd_cmp_test.go:13
  odo component delete, list and describe
  /go/src/github.com/openshift/odo/tests/integration/component.go:486
    should pass outside a odo directory with component name as parameter [It]
    /go/src/github.com/openshift/odo/tests/integration/component.go:525
     Expected
      
        <string>: {
          "kind": "Component",
          "apiVersion": "odo.openshift.io/v1alpha1",
          "metadata": {
            "name": "nodejs",
            "namespace": "cjmwpnlhak",
            "creationTimestamp": null
          },
          "spec": {
            "app": "app",
            "type": "nodejs",
            "source": "file://./",
            "env": [
              {
                "name": "DEBUG_PORT",
                "value": "5858"
              }
            ]
          },
          "status": {
            "state": "Pushed"
          }
        }
    to match JSON of
        <string>: {
          "kind": "Component",
          "apiVersion": "odo.openshift.io/v1alpha1",
          "metadata": {
            "name": "nodejs",
            "namespace": "cjmwpnlhak",
            "creationTimestamp": null
          },
          "spec": {
            "app": "app",
            "type": "nodejs",
            "source": "file://./",
            "env": [
              {
                "name": "DEBUG_PORT",
                "value": "5858"
              }
            ],
            "ports": [
              "8080/TCP"
            ]
          },
          "status": {
            "state": "Pushed"
          }
        }
        
    
    first mismatched key: "spec"
     /go/src/github.com/openshift/odo/tests/integration/component.go:541
```
Details - https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/2164/pull-ci-openshift-odo-master-v4.2-integration/258

**Comment by amitkrout** [comment]
This pr blocks rest pr in pool to go in 
ping @kadel @mohammedzee1000 @dharmit

**Comment by mohammedzee1000** [comment]
We have less choice atm as its blocking, @amitkrout could you update the issue tracking this that this is a quick fix and a more formal fix needs to be done :)

**Comment by amitkrout** [comment]
> We have less choice atm as its blocking, @amitkrout could you update the issue tracking this that this is a quick fix and a more formal fix needs to be done :)

There is no more formal fix i guess. I just combine those dependent test and make it a single scenario.

File: tests/integration/loginlogout/cmd_login_logout_test.go [hunk 1]
```diff
@@ -39,29 +39,27 @@ var _ = Describe("odo login and logout command tests", func() {
...
 			Expect(session1).To(ContainSubstring("odo project create <project-name>"))
 			session1 = oc.GetLoginUser()
 			Expect(session1).To(ContainSubstring(loginTestUserForNoProject))
-			// One initialization needs one login, hence it happens here
-			testUserToken1 = oc.GetToken()
-		})
 
-		It("Should login successfully with token without any projects with appropriate message", func() {
+			// Login successful with token without any projects with appropriate message
+			testUserToken1 = oc.GetToken()
```
Comment (mik-dass) [inline tests/integration/loginlogout/cmd_login_logout_test.go]: why are we getting the token twice?

Comment (amitkrout) [reply tests/integration/loginlogout/cmd_login_logout_test.go]: @mik-dass The variable name itself itself suggest that this token is for test user not for the first current user ```developer```.

testUserToken1 = oc.GetToken() gets the login token for [user](https://github.com/openshift/odo/blob/master/tests/integration/loginlogout/cmd_login_logout_test.go#L13) ```loginTestUserForNoProject```. 

currentUserToken1 = oc.GetToken() gets the login token for user ```developer```. It is the same ```developer``` user when we do the login for first time CI test run. For reference check https://github.com/openshift/odo/blob/master/.travis.yml#L35 and https://github.com/openshift/odo/blob/master/scripts/configure-installer-tests-cluster.sh#L86

Comment (amitkrout) [reply tests/integration/loginlogout/cmd_login_logout_test.go]: @mik-dass I have added comments. PTAL

**Comment by amitkrout** [comment]
```
[odo]  ✗  Waiting for build to finish [30s]
[odo]  ✗  Failed to create component with name wo-wait-javaee-git-test. Please use `odo config view` to view settings used to create component. Error: unable to wait for build wo-wait-javaee-git-test-app-1 to run: timed out waiting for the condition
```
/test v4.1-e2e-scenarios

**Pull request merged**